### PR TITLE
Digging gets the same progress bar as minning.

### LINKED ISSUE #10854: No progress overlay for digging
Exactly what it says on the tin: mining rock has progress overlays, but digging up sand on the asteroid does not.

I've only tested this with the standard pickaxe, not the shovel or advanced picks yet. Will do.

### PR DESCRIPTION
**Pull request by astralenigma**
Fixes #10854

### CODE REVIEW

**Comment by phil235** [comment]
It works. But you could remove the copypasta by doing:
```var/digging_speed = 0```
 ```if(istype(W, /obj/item/weapon/shovel))```
```___var/obj/item/weapon/shovel/S = W```
```___digging_speed = S.dig_speed```
```else if(istype(W, /obj/item/weapon/pickaxe))```
 ```___var/obj/item/weapon/pickaxe/P = W```
``___digging_speed = P.dig_speed``
``if(digging_speed)``
``___stuff goes here``

Also, no need to call "update_icon()" it's not needed as it doesn't do anything. You can also remove the update_icon() on line 437. There's also too many parenthesis on the istype pickaxe and istype shovel checks.

**Comment by astralenigma** [comment]
Done because you are right Phill.

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -579,13 +565,13 @@ var/global/list/rockTurfEdgeCache
...
-			user << "<span class='notice'>You dig a hole.</span>"
-			gets_dug()
-			return
-
+		
+		if(do_after(user, digging_speed, target = src))
+			if(istype(src, /turf/simulated/floor/plating/asteroid)) //sanity check against turf being deleted during digspeed delay
+				user << "<span class='notice'>You dig a hole.</span>"
+				gets_dug()
+				feedback_add_details("pick_used_mining","[W.name]")
```
Comment (Iamgoofball) [inline code/modules/mining/mine_turfs.dm]: :+1:

Comment (astralenigma) [inline code/modules/mining/mine_turfs.dm]: I tested to make sure I understanded what phil meant and it's working I'm happy.

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -548,27 +547,14 @@ var/global/list/rockTurfEdgeCache
...
-		sleep(20)
-		if ((user.loc == T && user.get_active_hand() == W))
-			user << "<span class='notice'>You dig a hole.</span>"
-			gets_dug()
-			return
-
-	if ((istype(W, /obj/item/weapon/pickaxe)))
+		var/obj/item/weapon/shovel/S = W
+		digging_speed = S.digspeed
+	else if ((istype(W, /obj/item/weapon/pickaxe)))
```
Comment (phil235) [inline code/modules/mining/mine_turfs.dm]: Still one pair of parenthesis too many

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -548,27 +547,14 @@ var/global/list/rockTurfEdgeCache
 	//note that this proc does not call ..()
 	if(!W || !user)
 		return 0
-
+	var/digging_speed = 0;
 	if ((istype(W, /obj/item/weapon/shovel)))
```
Comment (phil235) [inline code/modules/mining/mine_turfs.dm]: same here, one pair of parenthesis can be removed

File: code/modules/mining/mine_turfs.dm [hunk 4]
```diff
@@ -548,27 +547,14 @@ var/global/list/rockTurfEdgeCache
 	//note that this proc does not call ..()
 	if(!W || !user)
 		return 0
-
+	var/digging_speed = 0;
```
Comment (phil235) [inline code/modules/mining/mine_turfs.dm]: The `;` isn't needed.

Comment (astralenigma) [inline code/modules/mining/mine_turfs.dm]: Didn't even noticed putting it.

**Comment by Aranclanos** [comment]
the comment '//sanity check against turf being deleted during digspeed delay' isn't needed and same with the changelog entry

**Pull request merged**